### PR TITLE
dvc: ignore Stage.meta in md5 calculations plus some polish

### LINKED ISSUE #2209: stage: marked as changed after modifying meta attribute
`DVC version: 0.50.1`

The meta attribute should not affect the reproduction of a stage.

### Steps to reproduce

1. Save a DVC-file (_capitalize.dvc_) with the following content:

```yaml
cmd: cat input.txt | tr a-z A-Z > output.txt

deps:
- path: input.txt

outs:
- path: output.txt

meta:
  author: mroutis
```

2. Create _input.txt_: `echo hello > input.txt`

3. Reproduce it: `dvc repro capitalize.dvc`

4. Add another key to meta:
```diff
meta:
  author: mroutis
+ issue: 2209
```
5. Reproduce it again: `dvc repro capitalize.dvc`

### PR DESCRIPTION
**Pull request by Suor**
See commit messages for changes. Fixes #2209.

### CODE REVIEW

**Comment by efiop** [comment]
@Suor Tests failed.

File: dvc/utils/__init__.py [hunk 1]
```diff
@@ -88,32 +88,27 @@ def bytes_md5(byts):
...
+        return [dict_filter(e, exclude) for e in d]
 
-            assert isinstance(k, str)
-            if k in exclude:
-                continue
-            ret[k] = dict_filter(v, exclude)
-        return ret
+    elif isinstance(d, dict):
+        items = (
+            (str(k) if isinstance(k, builtin_str) else k, v)
```
Comment (efiop) [inline dvc/utils/__init__.py]: This did not become easier to read, not at all.

Comment (Suor) [reply dvc/utils/__init__.py]: How about now?

File: dvc/utils/__init__.py [hunk 2]
```diff
@@ -88,32 +88,28 @@ def bytes_md5(byts):
...
-        ret = []
-        for e in d:
-            ret.append(dict_filter(e, exclude))
-        return ret
-    elif isinstance(d, dict):
-        ret = {}
-        for k, v in d.items():
-            if isinstance(k, builtin_str):
-                k = str(k)
+        return [dict_filter(e, exclude) for e in d]
```
Comment (efiop) [inline dvc/utils/__init__.py]: Any point in refactoring this? Again, single-lines are not easy to read. Also, you are breaking git blame by it, right? :slightly_smiling_face:

Comment (Suor) [reply dvc/utils/__init__.py]: It all started from mutable default, which might be problematic. Many lines are not easy to read) We will change code, this is inevitable.

**Pull request merged**